WordPress/Classes/Services/RawBlockEditorSettingsService.swift

```diff
@@ -24,48 +25,50 @@ class RawBlockEditorSettingsService {
             guard let dictionary = response as? [String: Any] else {
                 throw NSError(domain: "RawBlockEditorSettingsService", code: 1, userInfo: [NSLocalizedDescriptionKey: "Invalid response format"])
             }
-            blog.rawBlockEditorSettings = dictionary
+            let objectID = TaggedManagedObjectID(blog)
+            try? await ContextManager.shared.performAndSave { context in
+                let blog = try context.existingObject(with: objectID)
+                blog.setBlockEditorSettings(dictionary)
+            }
             return dictionary
         case .failure(let error):
             throw error
         }
     }
 
+    /// Refreshes the editor settings in the background.
+    func refreshSettings() {
+        Task { @MainActor in
+            try? await fetchSettings()
+        }
+    }
+
     @MainActor
-    func fetchSettings() async throws -> [String: Any] {
-        // Start a background refresh if not already refreshing
-        if !isRefreshing {
-            isRefreshing = true
-            Task {
-                do {
-                    _ = try await fetchSettingsFromAPI()
-                } catch {
-                    DDLogError("Error refreshing block editor settings: \(error)")
-                }
-                isRefreshing = false
+    private func fetchSettings() async throws -> [String: Any] {
+        if let task = refreshTask {
```

Comment:
This is a standard pattern how you approach this in Swift Concurrency. `Task` works as a future and can have multiple subscribers.

Reply:
I see now how proper utilization of `Task` simplifies the implementation.